kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/persistence/proto/AbstractProtoGenerator.java

```diff
@@ -69,9 +73,25 @@ public Collection<GeneratedFile> generateProtoFiles() {
         return generatedFiles;
     }
 
+    private void validateClasses() {
+        Set<String> modelNames = modelClasses.stream()
+                .map(c -> extractName(c))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(String::toLowerCase)
+                .collect(Collectors.toSet());
+        dataClasses.forEach(c -> extractName(c).ifPresent(name -> {
+            if (modelNames.contains(name.toLowerCase())) {
+                throw new IllegalStateException(
+                        format("The class '%s' name, used as process variable, conflicts with the generated process model classes for Data Index protobuf. Please rename either the process '%s' or the Java class.",
```

Comment:
maybe
```suggestion
                        format("The data class '%s' name, used as process variable, conflicts with the generated process model classes for Data Index protobuf. Please rename either the process '%s' or the Java class.",
```
would provide a better context (data class vs. process model class)?

Reply:
+1